### PR TITLE
[No_Jira] update story and doc based on feedback

### PR DESCRIPTION
**Pull request by matteo-hertel**
Update the icon storybook and the docs based on the feedback received from the design point of view

### CODE REVIEW

File: native/packages/react-native-bpk-component-icon/stories.js [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 import React from 'react';
 import { storiesOf } from '@storybook/react-native';
-import { Platform, View, StyleSheet } from 'react-native';
+import { Platform, View, StyleSheet, Text } from 'react-native';
```
Comment (georgegillams) [inline native/packages/react-native-bpk-component-icon/stories.js]: Can we use `BpkText` for the story titles?
I realise that's kinda ironic as you made the exact opposite comment to me the other day 😆

Comment (matteo-hertel) [reply native/packages/react-native-bpk-component-icon/stories.js]: Point taken, in this one I don't have BpkText as devDependency, in your bit you had it 😄

File: native/packages/react-native-bpk-component-icon/stories.js [hunk 2]
```diff
@@ -42,13 +47,102 @@ const styles = StyleSheet.create({
...
+      />
+      <BpkIcon
+        style={styles.singleIcon}
+        icon="hotels"
+      />
+    </View>
+  </View>);
+}
+
+function getColouredIcons() {
```
Comment (k0nserv) [inline native/packages/react-native-bpk-component-icon/stories.js]: Any particular reason we aren't using an arrow function here?

Comment (matteo-hertel) [reply native/packages/react-native-bpk-component-icon/stories.js]: none, i didn't felt the need for it! it'll be more consistent with the rest of the file tough so I'll update, thanks

File: native/packages/react-native-bpk-component-icon/stories.js [hunk 3]
```diff
@@ -42,13 +48,96 @@ const styles = StyleSheet.create({
...
+  bottomMargin: {
+    marginBottom: spacingSm,
   },
 });
 
+/*
+ * the following functions should go on the bottom
+ * but eslint complains about it -_-
+ * @return
+ */
```
Comment (matthewdavidson) [inline native/packages/react-native-bpk-component-icon/stories.js]: > the following functions should go on the bottom but eslint complains about it -_-

Why should those functions go on the bottom?

Comment (matthewdavidson) [reply native/packages/react-native-bpk-component-icon/stories.js]: Otherwise i'd remove that comment :).

Comment (matteo-hertel) [reply native/packages/react-native-bpk-component-icon/stories.js]: oh good catch, well I was using regular functions before and es-lint was complaining about them being declared after the use, function hoisting is not a thing according to es-lint 😄

**Pull request merged**